tf2_ros/src/transform_listener.cpp

```diff
@@ -67,13 +67,64 @@ TransformListener::~TransformListener()
 }
 
 
+inline
+bool
+_make_authority_str(
+  const std::vector<rclcpp::TopicEndpointInfo> & endpoints,
+  const rmw_gid_t & pub_gid,
+  std::string & authority)
```

Comment:
This is a nit, but instead of an output `authority` variable, maybe we can have this function return `std::string`, with an empty string meaning "not found"?  I'm not sure if that API is better, so feel free to disregard this comment if you don't like it.